core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -286,6 +286,34 @@ private[spark] class TaskSchedulerImpl(
     }
   }
 
+  /**
+   * SPARK-25250: Whenever any Result Task gets successfully completed, we simply mark the
+   * corresponding partition id as completed in all attempts for that particular stage and
+   * additionally, for a Result Stage, we also kill the remaining task attempts running on the
+   * same partition. As a result, we do not see any Killed tasks due to
+   * TaskCommitDenied Exceptions showing up in the UI.
+   */
+  override def completeTasks(partitionId: Int, stageId: Int, killTasks: Boolean): Unit = {
+    taskSetsByStageIdAndAttempt.getOrElse(stageId, Map()).values.foreach { tsm =>
+      tsm.partitionToIndex.get(partitionId) match {
+        case Some(index) =>
+          tsm.markPartitionCompletedForRedundantTaskAttempts(index)
```

Comment:
hmmm, here, may looks like this:

```
if (!tsm.isZombie) {
  tsm.markPartitionCompleted 
} else {
 // leave zombie tasksets to do nothing
}
```

cc @cloud-fan , WDYT ?

Reply:
I may not agree with you on this one as there can also be a case where a non-zombie task might succeed first and we might then need to kill the corresponding zombie task. However, @cloud-fan , would appreciate your opinion on this one. Thank you.

Reply:
Ah, sorry. I forgot to mention killing sttuf.  Killing tasks is required whether it is a zombie or non-zombie ts manager.

Reply:
I agree with @pgandhi999